src/components/PaletteForm/PaletteForm.js

```diff
@@ -17,56 +19,94 @@ class PaletteForm extends Component {
     while (colors.length < 5) {
       colors.push(this.getRandomColor())
     }
+    colors = colors.map((color, i) => {
+      return { [`color${i + 1}`]: color, isLocked: false }
+    })
     this.setState({ colors })
   }
 
   getRandomColor() {
     return "#000000".replace(/0/g,() => {return (~~(Math.random()*16)).toString(16);});
   }
 
-  updateColors(e) {
-    let colors = [];
-    while (colors.length < 5) {
-      colors.push(this.getRandomColor())
-    }
+  updateColors = (e) => {
+    let { colors } = this.state;
+    colors = colors.map((color, i) => {
+      if (color.isLocked === false) {
+         return {
+          [`color${i + 1}`]: this.getRandomColor(), 
+          isLocked: false
+        }
```

Comment:
It would be nice to have a helper function here that checks the new color vs the rest of the other colors to ensure that the new color is not a duplicate of any of the old ones.